dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -814,8 +833,18 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
                     datafileSolrInputDocument.addField(SearchFields.CATEGORY_OF_DATAVERSE, dataset.getDataverseContext().getIndexableCategoryName());
                     datafileSolrInputDocument.addField(SearchFields.ACCESS,
                                                        fileMetadata.getTermsOfUse().getTermsOfUseType() == TermsOfUseType.RESTRICTED ? SearchConstants.RESTRICTED : SearchConstants.PUBLIC);
+                    
+                    if (!dataset.hasActiveEmbargo()) {
+                        if (fileMetadata.getTermsOfUse().getTermsOfUseType() == TermsOfUseType.LICENSE_BASED && fileMetadata.getTermsOfUse().getLicense() != null) {
```

Comment:
Suggestion: we can store `fileMetadata.getTermsOfUse().getTermsOfUseType()` or at least  `fileMetadata.getTermsOfUse()` in a variable, so we have more readable code here.